### PR TITLE
fauna reagent health

### PR DESCRIPTION
**Pull request by jjpark-kb**
## About The Pull Request
so, reagents typically are just useless on fauna/megafauna... I plan on changing that in some basic form at the moment.

## How This Contributes To The Skyrat Roleplay Experience
toxins (and medicines) should affect some fauna/megafauna as they are living beings... it's expected reagents would affect them.

## Changelog

:cl:
add: mining mobs and megafauna are now impacted slightly by toxins and medicine
/:cl:

### CODE REVIEW

**Comment by Whygil** [comment]
Isn't it for a balance? Bluespace artillery can't kill megafauna either. Same reaction on chemicals would be weird, considering that they are horrible monsters from hell.

**Comment by jjpark-kb** [comment]
> Isn't it for a balance? Bluespace artillery can't kill megafauna either. Same reaction on chemicals would be weird, considering that they are horrible monsters from hell.

mentioning balance, around me? hah.

**Review by Zonespace27 — COMMENTED** [review]
What's the overhead on adding reagents to all megas and asteroid mobs

File: modular_skyrat/modules/fauna_reagent/fauna_reagent.dm [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+/mob/living/simple_animal/hostile/megafauna
+	reagent_health = TRUE
+
+/mob/living/simple_animal/hostile/asteroid
+	reagent_health = TRUE
+
+/mob/living/simple_animal/Life(delta_time, times_fired)
+	. = ..()
+	if(reagent_health && reagents)
+		for(var/datum/reagent/reagents_within in reagents.reagent_list)
```
Comment (tf-4) [inline modular_skyrat/modules/fauna_reagent/fauna_reagent.dm]:
```suggestion
		for(var/datum/reagent/reagents_within as anything in reagents.reagent_list)
```

**Comment by jjpark-kb** [comment]
> What's the overhead on adding reagents to all megas and asteroid mobs

not sure, would be nice to have a TM to see overhead on the main server.

File: modular_skyrat/modules/fauna_reagent/fauna_reagent.dm (hunk 1 above)
Comment (jjpark-kb) [reply modular_skyrat/modules/fauna_reagent/fauna_reagent.dm]: how it is currently written works. I can test later if this is fine as well.

**Comment by tf-4** [comment]
Why does it need a TM to see that? I'm not sure what would differ on live versus local with this PR

**Comment by jjpark-kb** [comment]
you're right, a merge would be better. It literally won't be much at all since all human mobs already do this essentially.

File: modular_skyrat/modules/fauna_reagent/fauna_reagent.dm (hunk 1 above)
Comment (Zonespace27) [reply modular_skyrat/modules/fauna_reagent/fauna_reagent.dm]: above method is faster

**Pull request merged**